Formula/podman.rb

```diff
@@ -1,97 +1,229 @@
+    sha256 cellar: :any_skip_relocation, catalina:       "0f443433962a0fc1555da4e75550aae083075c66c438d38ef98dde1dbeb67eba"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "d64e188e78c05935c6d73713f69ffae54452082e01b9d3f8f7a64c160c302fe9"
+  end
 
-    # Allow specifying helper dirs with $BINDIR as base directory. Use a `$BINDIR` magic
-    # token as a prefix in the helper path to indicate it should be relative to the
-    # directory where the binary is located.
-    # This patch can be dropped on upgrade to podman 4.3.0.
-    patch do
-      url "https://github.com/containers/common/commit/030b7518103cfd7b930b54744d4a4510b659fdc2.patch?full_index=1"
-      sha256 "7c00abe7728d6438abcdb69ce6efa43503dcbb93bcb2d737f6ca4aa553e2eeb5"
-      directory "vendor/github.com/containers/common"
-    end
+  depends_on "go-md2man" => :build
 
-    # Update Darwin config to include '$BINDIR/../libexec/podman' in helper search path.
-    # This patch can be dropped on upgrade to podman 4.3.0.
-    patch do
-      url "https://github.com/containers/common/commit/4e6828877b0067557b435ec8810bda7f5cb48a4f.patch?full_index=1"
-      sha256 "01c4c67159f83f636a7b3a6007a010b1336c83246e4fb8c34b67be32fd6c2206"
-      directory "vendor/github.com/containers/common"
-    end
+  on_macos do
+    # Required latest gvisor.dev/gvisor/pkg/gohacks
+    # Try to switch to the latest go on the next release
+    depends_on "go@1.18" => :build
+    depends_on "qemu"
+  end
+
+  on_linux do
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "go" => :build
+    depends_on "pkg-config" => :build
+    depends_on "rust" => :build
+    depends_on "conmon"
+    depends_on "crun"
+    depends_on "fuse-overlayfs"
+    depends_on "gpgme"
+    depends_on "libseccomp"
+    depends_on "slirp4netns"
+    depends_on "systemd"
+  end
 
-    resource "gvproxy" do
+  resource "gvproxy" do
+    on_macos do
       url "https://github.com/containers/gvisor-tap-vsock/archive/v0.4.0.tar.gz"
       sha256 "896cf02fbabce9583a1bba21e2b384015c0104d634a73a16d2f44552cf84d972"
     end
   end
 
-  bottle do
-    sha256 cellar: :any_skip_relocation, arm64_monterey: "7a7443b447e60d3f96b523e16f6f87f2ce85373ca8d9225ae21b172fc95a1ea6"
-    sha256 cellar: :any_skip_relocation, arm64_big_sur:  "a56477a832e836895b1fa7f9847f80d4c3e55ef4b1409c8d14bee06d8b4c1e4d"
-    sha256 cellar: :any_skip_relocation, monterey:       "f5cd5677ba4ff64682eeecb8422ae58bba5327d28480960bc315a6cebea13167"
-    sha256 cellar: :any_skip_relocation, big_sur:        "6dd0b75f8496dfe0c3bb618d8194a2ce7e5521881383045385ecbc3225f2e007"
-    sha256 cellar: :any_skip_relocation, catalina:       "0f443433962a0fc1555da4e75550aae083075c66c438d38ef98dde1dbeb67eba"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:   "d64e188e78c05935c6d73713f69ffae54452082e01b9d3f8f7a64c160c302fe9"
+  resource "catatonit" do
```

Comment:
Not sure if those resources should be separate formulae instead. That would certainly make the podman formula easier to maintain.